### PR TITLE
Mark recently reindexed course metadata records

### PR DESCRIPTION
**Pull request by iloveagent57**
...so that subsequent batches of algolia reindexing don't reindex them a second time.

This is a minor but hopefully valuable optimization.  Because our algolia task looks for metadata records to update in batches, _and_ because it looks for records based on key **or** parent key, it's possible that we'll update the same course record in algolia multiple times across multiple batches within a single task.

The solution here is to mark recently-reindexed course metadata records in the cache and skip over them if they're present.

Why use the cache instead of a `set()` called `recently_updated_course_records` within the task body?
Answer: this is a little of a belt-and-suspenders approach.  In case we get into a situation where more task invocations are happening than we'd like, this is a sort-of failback, as long as those extra task invocations occur within the cache expiry period - which is 30 minutes in this PR.

### CODE REVIEW

File: enterprise_catalog/apps/api/tasks.py [hunk 1]
```diff
@@ -372,6 +374,9 @@ def _reindex_algolia(content_keys):
         course_content_metadata = content_metadata.filter(content_type=COURSE)
         for metadata in course_content_metadata:
             content_key = metadata.content_key
+            if _was_recently_indexed(content_key):
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/api/tasks.py:377]: nit: Would it make sense to filter out / exclude recently indexed content keys before we iterated through all `course_content_metadata`? I guess we'd be iterating through `course_content_metadata` twice at that point, though so maybe a moot point 🤷‍♂️

**Review by adamstankiewicz — APPROVED** [review]
I think the approach of using the django cache here is great! I agree that using the cache feels a bit safer than using a local `set()` within the task itself to handle the case you outlined in the PR description.

File: enterprise_catalog/apps/api/tasks.py (hunk 1 above)
Comment (iloveagent57) [reply enterprise_catalog/apps/api/tasks.py:377]: I like to make a kind of "cache sandwich" out of the code, so it's easier to tell what's going on.

cache sandwich, serves 1:
Top slice of bread ~= check if something's in the cache, maybe early return
Meat/veggies ~= do the work
Bottom slice of bread ~= put something in the cache

Comment (adamstankiewicz) [reply enterprise_catalog/apps/api/tasks.py:377]: Yeah, having the check whether something's in the cache nested within the iteration probably does help with readability / general understanding.

Also, side note: In a world with cached sandwiches, if something is already in the cache, is your sandwich literally just a slice of bread? 🍞 🥖

**Pull request merged**